packages/store/src/StoreCore.sol

```diff
@@ -533,16 +530,25 @@ library StoreCoreInternal {
     uint8 schemaIndex,
     bytes memory data
   ) internal {
-    // verify the value has the correct length for the field
-    SchemaType schemaType = valueSchema.atIndex(schemaIndex);
-    if (schemaType.getStaticByteLength() != data.length) {
-      revert IStoreErrors.StoreCore_InvalidDataLength(schemaType.getStaticByteLength(), data.length);
-    }
-
-    // Store the provided value in storage
     uint256 location = _getStaticDataLocation(tableId, key);
     uint256 offset = _getStaticDataOffset(valueSchema, schemaIndex);
+
     Storage.store({ storagePointer: location, offset: offset, data: data });
+
+    // Prepare data for the splice event
+    uint256 start = offset;
+    uint256 deleteCount = valueSchema.atIndex(schemaIndex).getStaticByteLength();
```

Comment:
i think we should use `data.length` here (since onchain all of `data` is written, independent of whether or not it matches the expected static byte length at this index)